src/vfs2.c

```diff
@@ -456,14 +456,13 @@ static int vfs2_read(sqlite3_file *file, void *buf, int amt, sqlite3_int64 ofst)
 	return orig->pMethods->xRead(orig, buf, amt, ofst);
 }
 
-static int wal_swap(struct entry *e, const struct wal_hdr *wal_hdr)
+/**
+ * Update the volatile state by exchanging WAL-cur and WAL-prev.
+ */
+static void wal_swap(struct entry *e, const struct wal_hdr *hdr)
 {
-	PRE(e->pending_txn_len == 0);
-	PRE(e->pending_txn_frames == NULL);
```

Comment:
My thought was that this should be the responsibility of the wtx invariant instead of ending up as an ad-hoc assertion. I haven't re-added the invariant yet, maybe I should do that before merging this PR?